packages/features/bookings/lib/getBookingFields.ts

```diff
@@ -0,0 +1,245 @@
+  };
+
+  // These fields should be added before other user fields
+  const systemBeforeFields: typeof bookingFields = [
+    {
+      defaultLabel: "your_name",
+      defaultPlaceholder: "example_name",
+      type: "name",
+      name: "name",
+      required: true,
+      sources: [
+        {
+          label: "Default",
+          id: "default",
+          type: "default",
+        },
+      ],
+    },
+    {
+      defaultLabel: "email_address",
+      defaultPlaceholder: "you@example.com",
+      type: "email",
+      name: "email",
+      required: true,
+      sources: [
+        {
+          label: "Default",
+          id: "default",
+          type: "default",
+        },
+      ],
+    },
+    {
+      defaultLabel: "location",
+      type: "radioInput",
+      name: "location",
+      // Even though it should be required it is optional in production with backend choosing CalVideo as the fallback
+      required: false,
+      // Populated on the fly from locations. I don't want to duplicate storing locations and instead would like to be able to refer to locations in eventType.
+      // options: `eventType.locations`
+      optionsInputs: {
+        attendeeInPerson: {
+          type: "address",
+          required: true,
+          placeholder: "",
+        },
+        phone: {
+          type: "phone",
+          required: true,
+          placeholder: "",
+        },
+      },
+      sources: [
+        {
+          label: "Default",
+          id: "default",
+          type: "default",
+        },
+      ],
+    },
+  ];
+
+  // These fields should be added after other user fields
+  const systemAfterFields: typeof bookingFields = [
+    {
+      defaultLabel: "additional_notes",
+      type: "textarea",
+      name: "notes",
+      required: additionalNotesRequired,
+      defaultPlaceholder: "share_additional_notes",
+      sources: [
+        {
+          label: "Default",
+          id: "default",
+          type: "default",
+        },
+      ],
+    },
+    {
+      defaultLabel: "additional_guests",
+      type: "multiemail",
+      name: "guests",
+      required: false,
+      hidden: disableGuests,
+      sources: [
+        {
+          label: "Default",
+          id: "default",
+          type: "default",
+        },
+      ],
+    },
+    {
+      defaultLabel: "reschedule_reason",
+      type: "textarea",
+      name: "rescheduleReason",
+      defaultPlaceholder: "reschedule_placeholder",
+      required: false,
+      sources: [
+        {
+          label: "Default",
+          id: "default",
+          type: "default",
+        },
+      ],
+    },
+  ];
+
+  const missingSystemBeforeFields = [];
+  for (const field of systemBeforeFields) {
+    // Only do a push, we must not update existing system fields as user could have modified any property in it,
+    if (!bookingFields.find((f) => f.name === field.name)) {
+      missingSystemBeforeFields.push(field);
+    }
+  }
```

Comment:
NIT: I believe this could be a map (would save milliseconds)  / look a little cleaner

Reply:
Makes sense :)